yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -17,26 +18,48 @@ import {
   initExecutionEnvironment,
   initGlobalVariables,
   initL1ToL2MessageOracleInput,
+  initMachineState,
 } from './fixtures/index.js';
-import { Add, CalldataCopy, Return } from './opcodes/index.js';
+import { Add, CalldataCopy, Instruction, Return } from './opcodes/index.js';
 import { encodeToBytecode } from './serialization/bytecode_serialization.js';
 
 describe('AVM simulator: injected bytecode', () => {
-  it('Should execute bytecode that performs basic addition', async () => {
-    const calldata: Fr[] = [new Fr(1), new Fr(2)];
+  let calldata: Fr[];
+  let ops: Instruction[];
+  let bytecode: Buffer;
 
-    // Construct bytecode
-    const bytecode = encodeToBytecode([
+  beforeAll(() => {
+    calldata = [new Fr(1), new Fr(2)];
+    ops = [
       new CalldataCopy(/*indirect=*/ 0, /*cdOffset=*/ adjustCalldataIndex(0), /*copySize=*/ 2, /*dstOffset=*/ 0),
       new Add(/*indirect=*/ 0, TypeTag.FIELD, /*aOffset=*/ 0, /*bOffset=*/ 1, /*dstOffset=*/ 2),
       new Return(/*indirect=*/ 0, /*returnOffset=*/ 2, /*copySize=*/ 1),
-    ]);
+    ];
+    bytecode = encodeToBytecode(ops);
+  });
 
+  it('Should execute bytecode that performs basic addition', async () => {
     const context = initContext({ env: initExecutionEnvironment({ calldata }) });
+    const { l2GasLeft: initialL2GasLeft } = AvmMachineState.fromState(context.machineState);
     const results = await new AvmSimulator(context).executeBytecode(bytecode);
+    const expectedL2GasUsed = ops.reduce((sum, op) => sum + op.gasCost().l2Gas, 0);
 
     expect(results.reverted).toBe(false);
     expect(results.output).toEqual([new Fr(3)]);
+    expect(expectedL2GasUsed).toBeGreaterThan(0);
```

Comment:
I'd rather have hardcoded numbers for these. Having code compute the expected outputs IMO should be avoided when possible.

Reply:
I didn't want the tests to break when we updated gas costs for those instructions, but agree on the sentiment of hardcoding. I'll do the change!